source/extensions/quic_listeners/quiche/platform/quic_mem_slice_impl.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include <iostream>
+#include <memory>
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/assert.h"
+
+#include "quiche/quic/core/quic_buffer_allocator.h"
+
+namespace quic {
+
+class QuicMemSliceImpl {
+public:
+  // Constructs an empty QuicMemSliceImpl.
+  QuicMemSliceImpl() = default;
+
+  // Constructs a QuicMemSliceImpl by let |allocator| allocate a data buffer of
+  // |length|.
+  QuicMemSliceImpl(QuicBufferAllocator* allocator, size_t length)
+      : length_(length), slice_(new Envoy::Buffer::OwnedImpl()) {
+    auto fragment = new Envoy::Buffer::BufferFragmentImpl(
```

Comment:
There are no current references to BufferFragmentImpl in any .cc file under //source/.., making me wonder if this is the right interface for you to use.

I think most uses of Buffer start with instantiating Buffer::OwnedImpl. Would it be possible for you to refactor to use that, rather than coming in at this level?

You could argue that if no one outside the buffer infrastucture uses Buffer::BufferFragmentImpl that it shouldn't be public, and I'd be inclined to agree :)

@brian-pane do you have thoughts here?